advcpmv-0.9-8.32.patch

```diff
@@ -0,0 +1,702 @@
+@@ -304,4 +307,17 @@
+ bool chown_failure_ok (struct cp_options const *) _GL_ATTRIBUTE_PURE;
+ mode_t cached_umask (void);
+ 
++/* BEGIN progress mod */
++int countfiles( const char *path );
++long getDirSize( const char *dirname );
++int file_size_format ( char * _cDst, long _lSize, int _iCounter );
++
++__attribute__((__common__)) long g_iTotalSize;
++__attribute__((__common__)) long g_iTotalWritten;
++__attribute__((__common__)) int g_iFilesCopied;
++__attribute__((__common__)) struct timeval g_oStartTime;
++__attribute__((__common__)) int g_iTotalFiles;
++__attribute__((__common__)) bool progress;
++/* END progress mod */
++
+ #endif
+diff -aur coreutils-8.32/src/cp.c coreutils-8.32-patched/src/cp.c
+--- coreutils-8.32/src/cp.c	2020-01-01 15:13:12.000000000 +0100
++++ coreutils-8.32-patched/src/cp.c	2021-02-16 23:31:38.495644611 +0100
+@@ -131,6 +131,7 @@
+   {"symbolic-link", no_argument, NULL, 's'},
+   {"target-directory", required_argument, NULL, 't'},
+   {"update", no_argument, NULL, 'u'},
++  {"progress-bar", no_argument, NULL, 'g'},
+   {"verbose", no_argument, NULL, 'v'},
+   {GETOPT_SELINUX_CONTEXT_OPTION_DECL},
+   {GETOPT_HELP_OPTION_DECL},
+@@ -170,6 +171,7 @@
+   -f, --force                  if an existing destination file cannot be\n\
+                                  opened, remove it and try again (this option\n\
+                                  is ignored when the -n option is also used)\n\
++  -g, --progress-bar           add a progress bar\n\
+   -i, --interactive            prompt before overwrite (overrides a previous -n\
+ \n\
+                                   option)\n\
+@@ -635,6 +637,42 @@
+         die (EXIT_FAILURE, 0, _("target %s is not a directory"),
+              quoteaf (file[n_files - 1]));
+     }
++    /* BEGIN progress mod */
++    struct timeval start_time;
++    if (progress) {
++        g_iTotalSize = 0;
++        g_iTotalFiles = 0;
++        g_iFilesCopied = 0;
++        g_iTotalWritten = 0;
++
++        /* save time */
++        gettimeofday ( & start_time, NULL );
++        g_oStartTime = start_time;
++
++        printf ( "Calculating total size... \r" );
++        fflush ( stdout );
++        long iTotalSize = 0;
++        int iFiles = n_files;
++        if ( ! target_directory )
++          iFiles = n_files - 1;
++        int j;
++
++        /* how many files are we copying */
++        g_iTotalFiles = countfiles (file[0]);
++        if ( iFiles > g_iTotalFiles )
++          g_iTotalFiles=iFiles;    
++
++        for (j = 0; j < iFiles; j++)
++        {
++          /* get size for each file */
++          iTotalSize += getDirSize(file[j]);
++          printf ( "Calculating total size... %ld\r", iTotalSize );
++          fflush ( stdout );
++        }
++        g_iTotalSize = iTotalSize;
++    }
++    /* END progress mod */
++
+ 
+   if (target_directory)
+     {
+@@ -777,6 +815,45 @@
+       ok = copy (source, new_dest, 0, x, &unused, NULL);
+     }
+ 
++    /* BEGIN progress mod */
```

Comment:
moved comment from after if  (original patch 0.8-8.32, line 463) and remove comment to remove everything, which made no sense to me?